### PR TITLE
Grant LPSRE backplane permissions for hypershift management clusters

### PR DESCRIPTION
**Pull request by thetechnick**
Signed-off-by: Nico Schieder <nschieder@redhat.com>

### What type of PR is this?
_(bug/feature/cleanup/documentation)_

### Which Jira/Github issue(s) this PR fixes?

_Fixes #MTSRE-874_

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Included documentation changes with PR
- [ ] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

File: deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre-management-cluster-project.ClusterRole.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  name: backplane-lpsre-management-cluster-project
+rules:
+- apiGroups:
+  - package-operator.run
+  resources: ['*']
+  verbs:
+  - get
+  - list
+  - watch
+  - update
```
Comment (jewzaam) [inline deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre-management-cluster-project.ClusterRole.yml:13]:
```suggestion
  - patch
  - update
```
Is this missed? Typically would include along with `update`.

File: deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre-management-cluster-project.ClusterRole.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - create
```
Comment (jewzaam) [inline deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre-management-cluster-project.ClusterRole.yml:14]: I would leave this out if not also granting `delete`.  Else you're stuck in how you can remediate incorrect creations.  My recommendation is use elevated access for both create and delete, these should not be typical for supporting?

File: deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre.SubjectPermission.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  name: backplane-lpsre-management-cluster
+  namespace: openshift-rbac-permissions
+spec:
+  permissions:
+  - allowFirst: true
+    clusterRoleName: admin
+    namespacesAllowedRegex: '(^openshift-package-operator$)'
+  - allowFirst: true
+    clusterRoleName: backplane-lpsre-management-cluster-project
+    namespacesAllowedRegex: '(^ocm-staging-.*|^ocm-production-.*|^hc-mc-.*|^openshift-package-operator$)'
```
Comment (jewzaam) [inline deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre.SubjectPermission.yml:13]: What are the `^hc-ms-.*` namespaces?

Why include environment in the ocm regex and not leave it open?  This misses integration.

**Review by jewzaam — CHANGES_REQUESTED** [review]
Changes requested and some question.  Generally +1 so adding approval now.

/approve

File: deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre.SubjectPermission.yml (hunk 3 above)
Comment (thetechnick) [reply deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre.SubjectPermission.yml]: I wanted to grant access to HCP namespaces, but I got the wrong prefixes.
Changed to `cluster-.*` which should make more sense.

Comment (jewzaam) [reply deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre.SubjectPermission.yml]: HCP NS are ocm-.*

Comment (thetechnick) [reply deploy/backplane/lpsre/hypershift/management-cluster/10-lpsre.SubjectPermission.yml]: Ok fixed. Thanks!
I have gotten mixed messages about that namespace prefix :)

**Pull request merged**